cli/fossilize_replay.cpp

```diff
@@ -2822,20 +2833,36 @@ struct ThreadedReplayer : StateCreatorInterface
 			                 }});
 
 			work.push_back({ get_order_index(ENQUEUE_OUT_OF_RANGE_PARENT_PIPELINES),
-			                 [this, &pipelines, derived]() {
+			                 [this, &pipelines, derived, &parents, parsed_parents]() {
 				                 // Figure out which of the parent pipelines we need.
 				                 for (auto &d : *derived)
-					                 enqueue_parent_pipelines(d, pipelines);
+					                 enqueue_parent_pipelines(d, pipelines, parents, *parsed_parents, 1);
 			                 }});
 
 			if (memory_index == 0)
 			{
 				// This is a join-like operation. We need to wait for all parent pipelines and all shader modules to have completed.
 				work.push_back({get_order_index(ENQUEUE_SHADER_MODULE_SECONDARY_OFFSET),
-				                [this, &parents, hash_offset, start_index]()
+				                [this, &pipelines, derived, &parents, parsed_parents, hash_offset, start_index]()
 				                {
-					                // Wait until all parent pipelines have been parsed.
-					                sync_worker_memory_context(PARENT_PIPELINE_MEMORY_CONTEXT);
+					                while (true)
+					                {
+						                // Wait until all parent pipelines have been parsed.
+						                sync_worker_memory_context(PARENT_PIPELINE_MEMORY_CONTEXT);
+
+						                unordered_set<Hash> dependencies = *parsed_parents;
```

Comment:
Would it make more sense to leave `dependencies` outside the loop and do `dependencies.clear(); std::swap(*parsed_parents, dependencies)` instead? Should avoid the deep copy.